bundles/org.openhab.binding.sonnen/src/main/java/org/openhab/binding/sonnen/internal/SonnenHandler.java

```diff
@@ -101,13 +112,20 @@ public void initialize() {
      * @return true if the update succeeded, false otherwise
      */
     private boolean updateBatteryData() {
-        String error = serviceCommunication.refreshBatteryConnection();
+        String error = serviceCommunication.refreshBatteryConnection(newAPI);
         if (error.isEmpty()) {
             if (!ThingStatus.ONLINE.equals(getThing().getStatus())) {
                 updateStatus(ThingStatus.ONLINE);
+                disconnectionCounter = 0;
             }
         } else {
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, error);
+            disconnectionCounter++;
+            if (disconnectionCounter > 60) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, error);
+            } else {
+                updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.COMMUNICATION_ERROR, error);
```

Comment:
This is not a valid combination. If actual reason is communication error, this should be `ThingStatus.OFFLINE` as well. For reference: https://www.openhab.org/docs/concepts/things.html#status-details

Reply:
ok, will change it with the next commit